adcc/adc_pp/state2state_transition_dm.py

```diff
@@ -20,12 +20,119 @@
+    return dm
+
+
+def s2s_tdm_adc2(mp, amplitude_l, amplitude_r, intermediates):
+    dm = s2s_tdm_adc0(mp, amplitude_l, amplitude_r, intermediates)
+
+    x1 = amplitude_l["s"]
+    y1 = amplitude_r["s"]
+    x2 = amplitude_l["d"]
+    y2 = amplitude_r["d"]
+
+    t2 = mp.t2(b.oovv)
+    p0_ov = mp.mp2_diffdm[b.ov]
+    p0_oo = mp.mp2_diffdm[b.oo]
+    p0_vv = mp.mp2_diffdm[b.vv]
+    p1_oo = dm[b.oo].evaluate()  # ADC(1) tdm
+    p1_vv = dm[b.vv].evaluate()  # ADC(1) tdm
+
+    rx1 = einsum('ijab,jb->ia', t2, x1)
+    ry1 = einsum('ijab,jb->ia', t2, y1)
+
+    dm[b.oo] = (
+        p1_oo - 2.0 * einsum('ikab,jkab->ij', y2, x2) + (
+            + 0.5 * (
+                einsum('ik,kj->ij', p1_oo, p0_oo)
+                + einsum('ik,kj->ij', p0_oo, p1_oo)
+            )
+            - einsum(
+                'ikcd,jkcd->ij', t2,
+                einsum('lk,jlcd->jkcd', 0.5 * p1_oo, t2)
+                - einsum('jkcb,db->jkcd', t2, p1_vv)
+            )
+            - 0.5 * (
+                einsum('ia,ja->ij', y1, einsum('jkac,kc->ja', t2, rx1))
+                + einsum('ikac,kc,ja->ij', t2, ry1, x1)
+            )
+            - einsum('ia,ja->ij', rx1, ry1)
+        ).symmetrise()
+    )
+    dm[b.vv] = (
+        p1_vv + 2.0 * einsum('ijac,ijbc->ab', x2, y2)
+        + (
+            - 0.5 * (
+                einsum("ac,cb->ab", p1_vv, p0_vv)
+                + einsum("ac,cb->ab", p0_vv, p1_vv)
+            )
+            - einsum(
+                "klbc,klac->ab", t2,
+                0.5 * einsum("klad,cd->klac", t2, p1_vv)
+                - einsum("jk,jlac->klac", p1_oo, t2)
+            )
+            + 0.5 * (
+                einsum("ikac,kc,ib->ab", t2, rx1, y1)
+                + einsum(
+                    "ia,ib->ab", x1,
+                    einsum("ikbc,kc->ib", t2, ry1)
+                )
+            )
+            + einsum("ia,ib->ab", ry1, rx1)
+        ).symmetrise()
+    )
+
+    p1_ov = -2.0 * einsum("jb,ijab->ia", x1, y2)
+    p1_vo = -2.0 * einsum("ijab,jb->ai", x2, y1)
+
+    dm[b.ov] = (
+        p1_ov
+        - einsum("ijab,bj->ia", t2, p1_vo)
+        - einsum("ib,ba->ia", p0_ov, p1_vv)
+        + einsum("ij,ja->ia", p1_oo, p0_ov)
+        - einsum(
+            "ib,ab->ia", y1,
+            einsum("klca,klcb->ab", t2, x2)
+        )
+        - einsum("ikcd,jkcd,ja->ia", t2, x2, y1)
+    )
+    dm[b.vo] = (
+        p1_vo
+        - einsum("ijab,jb->ai", t2, p1_ov)
+        - einsum("ib,ab->ai", p0_ov, p1_vv)
+        + einsum("ji,ja->ai", p1_oo, p0_ov)
+        - einsum(
+            "ib,ab->ai", x1,
+            einsum("klca,klcb->ab", t2, y2)
+        )
+        - einsum("ikcd,jkcd,ja->ai", t2, y2, x1)
+    )
+    return dm
+
+
+# Ref: https://doi.org/10.1080/00268976.2013.859313
+DISPATCH = {"adc0": s2s_tdm_adc0,
+            "adc1": s2s_tdm_adc0,       # same as ADC(0)
+            "adc2": s2s_tdm_adc2,
+            "adc2x": s2s_tdm_adc2,
+            "adc3": s2s_tdm_adc2,
```

Comment:
Really? There are no terms in ADC(3)?

Reply:
My bad... we only have ISR(2), but of course the mapping to ADC(3/2) properties happens one layer above 😄
I'm going to remove the `adc3` key here.